src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/graphs/KBestHaplotypeFinder.java

```diff
@@ -50,66 +52,66 @@ public KBestHaplotypeFinder(final SeqGraph graph, final Set<SeqVertex> sources,
+        final List<KBestHaplotype<V, E>> result = new ArrayList<>();
+        final PriorityQueue<KBestHaplotype<V, E>> queue = new PriorityQueue<>(Comparator.comparingDouble(KBestHaplotype<V, E>::score).reversed());
+        sources.forEach(source -> queue.add(new KBestHaplotype<>(source, graph)));
 
-        final Map<SeqVertex, MutableInt> vertexCounts = graph.vertexSet().stream()
+        final Map<V, MutableInt> vertexCounts = graph.vertexSet().stream()
                 .collect(Collectors.toMap(v -> v, v -> new MutableInt(0)));
 
         while (!queue.isEmpty() && result.size() < maxNumberOfHaplotypes) {
-            final KBestHaplotype pathToExtend = queue.poll();
-            final SeqVertex vertexToExtend = pathToExtend.getLastVertex();
+            final KBestHaplotype<V, E> pathToExtend = queue.poll();
+            final V vertexToExtend = pathToExtend.getLastVertex();
             if (sinks.contains(vertexToExtend)) {
                 result.add(pathToExtend);
             } else {
                 if (vertexCounts.get(vertexToExtend).getAndIncrement() < maxNumberOfHaplotypes) {
-                    final Set<BaseEdge> outgoingEdges = graph.outgoingEdgesOf(vertexToExtend);
+                    final Set<E> outgoingEdges = graph.outgoingEdgesOf(vertexToExtend);
                     int totalOutgoingMultiplicity = 0;
                     for (final BaseEdge edge : outgoingEdges) {
                         totalOutgoingMultiplicity += edge.getMultiplicity();
                     }
 
-                    for (final BaseEdge edge : outgoingEdges) {
-                        final SeqVertex targetVertex = graph.getEdgeTarget(edge);
-                        queue.add(new KBestHaplotype(pathToExtend, edge, totalOutgoingMultiplicity));
+                    for (final E edge : outgoingEdges) {
+                        final V targetVertex = graph.getEdgeTarget(edge);
+                        queue.add(new KBestHaplotype<>(pathToExtend, edge, totalOutgoingMultiplicity));
                     }
                 }
             }
         }
         return result;
     }
 
-    public List<KBestHaplotype> findBestHaplotypes() {
+    public List<KBestHaplotype<V, E>> findBestHaplotypes() {
        return findBestHaplotypes(Integer.MAX_VALUE);
     }
 
     /**
      * Removes edges that produces cycles and also dead vertices that do not lead to any sink vertex.
      * @return never {@code null}.
      */
-    private static SeqGraph removeCyclesAndVerticesThatDontLeadToSinks(final SeqGraph original, final Collection<SeqVertex> sources, final Set<SeqVertex> sinks) {
-        final Set<BaseEdge> edgesToRemove = new HashSet<>(original.edgeSet().size());
-        final Set<SeqVertex> vertexToRemove = new HashSet<>(original.vertexSet().size());
+    private BaseGraph<V, E> removeCyclesAndVerticesThatDontLeadToSinks(final BaseGraph<V, E> original, final Collection<V> sources, final Set<V> sinks) {
```

Comment:
Is it no longer `static` because Java has some issue with generic statics?  Just checking.

Reply:
Yes